pkg/operator/defragcontroller/defragcontroller.go

```diff
@@ -197,31 +208,26 @@ func (c *DefragController) checkDefrag(ctx context.Context, recorder events.Reco
 	return v1helpers.NewMultiLineAggregate(errors)
 }
 
-func (c *DefragController) ensureControllerDisabledCondition(ctx context.Context, controlPlaneTopology configv1.TopologyMode, recorder events.Recorder) error {
-	// Defrag is blocking and can only be safely performed in HighlyAvailableTopologyMode
-	if controlPlaneTopology == configv1.HighlyAvailableTopologyMode {
-		_, _, updateErr := v1helpers.UpdateStatus(ctx, c.operatorClient,
-			v1helpers.UpdateConditionFn(operatorv1.OperatorCondition{
-				Type:   defragDisabledCondition,
-				Status: operatorv1.ConditionFalse,
-				Reason: "AsExpected",
-			}))
-		if updateErr != nil {
-			recorder.Warning("DefragControllerUpdatingStatus", updateErr.Error())
-			return updateErr
-		}
-	} else {
+func (c *DefragController) ensureControllerDisabledCondition(ctx context.Context, desiredStatus operatorv1.ConditionStatus, recorder events.Recorder) error {
+	_, currentState, _, err := c.operatorClient.GetOperatorState()
+	if err != nil {
+		return err
+	}
+
+	controllerDisabledCondition := v1helpers.FindOperatorCondition(currentState.Conditions, defragDisabledCondition)
+	if controllerDisabledCondition == nil || controllerDisabledCondition.Status != desiredStatus {
```

Comment:
I think `v1helpers.UpdateStatus()` will already check if `current status == desired status` before updating so we could just call that directly without this check.

https://github.com/openshift/cluster-etcd-operator/blob/c7ed7ef9c4252feb588db290f84ae63577626dc7/vendor/github.com/openshift/library-go/pkg/operator/v1helpers/helpers.go#L163-L179

Reply:
very nice, thanks for telling me. fixed

Reply:
until I ran the unit tests :) apparently the mocked version does not have the same check so the unit tests were failing. Shall we keep it for now? I'm too lazy to get this patched in library-go